File: AI/src/state/MarbleColor.cpp
#include "MarbleColor.hpp"

MarbleColor getMarbleColorFromString(const std::string marbleColorString) {
    if (marbleColorString == "RED") {
        return MarbleColor::RED;
    }
    if (marbleColorString == "BLUE") {
        return MarbleColor::BLUE;
    }
    if (marbleColorString == "YELLOW") {
        return MarbleColor::YELLOW;
    }
    if (marbleColorString == "BLACK") {
        return MarbleColor::BLACK;
    }
    return MarbleColor::NONE;
}

File: GUI/src/states/MarbleColor.ts
export enum MarbleColor {
    RED = 'RED',
    BLUE = 'BLUE',
    YELLOW = 'YELLOW',
    BLACK = 'BLACK',
    NONE = 'NONE',
}

export const stringToMarbleColor = (colorString: string): MarbleColor => {
    const matchedMarbleColor = Object.values(MarbleColor).filter(
        value => value === colorString,
    );

    return matchedMarbleColor.length > 0
        ? matchedMarbleColor[0]
        : MarbleColor.NONE;
};
